source/common/http/async_client_impl.h

```diff
@@ -299,6 +299,9 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     ASSERT(buffered_body_ != nullptr);
   }
   const Buffer::Instance* decodingBuffer() override { return buffered_body_.get(); }
+  void modifyDecodingBuffer(std::function<void(Buffer::Instance&)> callback) override {
+    callback(*buffered_body_.get());
```

Comment:
maybe just NOT_IMPLEMENTED this and below? I don't think the router filter should ever call this currently?